From 354bee66dc6586589d81617074054edfc32cf618 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 9 Oct 2018 08:50:42 +0300
Subject: [PATCH 4801/5242] MLK-19845: drm/imx/dcss: fix suspend/resume issue

commit  07be4bc88bdb53edaf6266c7c8669ef30351fbf9 from
https://source.codeaurora.org/external/imx/linux-imx.git

While running suspend/resume tests it may happen to go to suspend while
CTXLD still has entries to be commited. Currently, when this happens,
the scaler freezes.

This patch will fire up context loader just before going to suspend,
thus commiting everything to DCSS before cutting off the clocks.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-ctxld.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/imx/dcss/dcss-ctxld.c b/drivers/gpu/imx/dcss/dcss-ctxld.c
index 248c68c..3162f7c 100644
--- a/drivers/gpu/imx/dcss/dcss-ctxld.c
+++ b/drivers/gpu/imx/dcss/dcss-ctxld.c
@@ -451,6 +451,8 @@ int dcss_ctxld_suspend(struct dcss_soc *dcss)
 	int wait_time_ms = 0;
 	unsigned long flags;
 
+	dcss_ctxld_kick(dcss);
+
 	while (ctxld->in_use && wait_time_ms < 500) {
 		msleep(20);
 		wait_time_ms += 20;
-- 
1.7.9.5

